### ISSUE TITLE
constant Tilt::Cache is deprecated

### ISSUE THREAD

**Issue by oehlschl** [jeremyevans/tilt#6]
> Deprecate Tilt::Cache (jeremyevans)

from https://github.com/jeremyevans/tilt/discussions/5#discussion-5267721

Is there an upgrade path or recommended alternative for projects currently using Tilt::Cache? For example, it's part of Sinatra, so I think others are likely to see and ask about this soon.
```
/vendor/bundle/ruby/3.1.0/gems/sinatra-3.0.6/lib/sinatra/base.rb:931: warning: constant Tilt::Cache is deprecated
```

Thanks in advance!

**Comment by jeremyevans** [comment]
I submitted a pull request to Sinatra to fix this issue last month, which was merged: https://github.com/sinatra/sinatra/pull/1922 . Until Sinatra does another release, I would recommend using the Sinatra main branch.

The easiest upgrade path for other projects using Tilt::Cache, would be to copy the code from Tilt and use it, though projects should really switch to something better.  I deprecated Tilt::Cache not just because Tilt didn't use it internally, but also because it is not thread safe and has other issues.  See Tilt::Cache's documentation for why it may be a bad idea to use it:  https://github.com/jeremyevans/tilt/blob/master/lib/tilt.rb#L100-L108
